rasa/importers/autoconfig.py

```diff
@@ -0,0 +1,115 @@
+import logging
+import os
+from pathlib import Path
+
+from typing import Text, Dict, Any, List
+
+from rasa.cli import utils as cli_utils
+from rasa.constants import CONFIG_AUTOCONFIGURABLE_KEYS
+from rasa.utils import io as io_utils
+
+logger = logging.getLogger(__name__)
+
+
+# In future iterations, this function and the function `create_config_for_keys` below
+# will get additional input parameters to allow for smarter configuration. E.g.
+# training_data: TrainingData
+# domain: Domain
+# stories: StoryGraph
+def get_auto_configuration(config_file) -> Dict[Text, Any]:
+    """Determine configuration from a configuration file.
+
+    Keys that are provided in the file are kept. Keys that are not provided are
+    configured automatically.
+    """
+    if config_file and os.path.exists(config_file):
+        config = io_utils.read_config_file(config_file)
+
+        missing_keys = [k for k in CONFIG_AUTOCONFIGURABLE_KEYS if not config.get(k)]
+        create_config_for_keys(config, missing_keys)
+
+        dump_config(config, config_file)
+    else:
+        config = {}
+
+    return config
+
+
+def create_config_for_keys(config: Dict[Text, Any], keys: List[Text]) -> None:
+    """Complete a config by adding automatic configuration for the specified keys.
+
+    Args:
+        config: The provided configuration.
+        keys: Keys to be configured automatically (e.g. `policies`, `pipeline`).
+    """
+    import pkg_resources
+
+    if keys:
+        logger.debug(
+            f"The provided configuration does not contain the key(s) {keys}. "
+            f"Running automatic configuration for them now."
+        )
+
+    default_config_file = pkg_resources.resource_filename(
+        __name__, "default_config.yml"
+    )
+
+    default_config = io_utils.read_config_file(default_config_file)
+
+    for key in keys:
+        config[key] = default_config[key]
+        # maybe add a debug output to print the result of the autoconfiguration?
+
+
+def dump_config(config: Dict[Text, Any], config_file: Text) -> None:
```

Comment:
It doesn't look like as if the additional configuration is dumped in a commented out way as proposed in https://docs.google.com/document/d/1bgmgr3tv3-HJ7l2Sl6DUR-pmU6C_PA7ypMs12kSRNig/edit?disco=AAAAGlk_9eI

the idea was that the auto configuration is run on every call to `rasa train`, if we dump the configuration as a normal keys to the file that won't happen as in the next run that configuration will be read instead of a new auto configuration.

using comments should make this easier as well as you can prepare the commented out section separately and should be able to replace any existing autoconfiguration comment with it based on the marker

Reply:
Yes, that's what I'm currently working on.

For the time being, it doesn't matter much (see here: https://github.com/RasaHQ/rasa/pull/5890#issuecomment-635888538), but we should have it being commented out before we release the feature. Tobi pointed out that even in v1 that's useful because we might at some point change the default config, and those changes won't get picked up if the auto config doesn't run again.

Reply:
yes exactly - perfect. just wanted to make sure we are on the same page :+1: